src/node_api.cc

```diff
@@ -447,8 +448,17 @@ class ThreadSafeFunction : public node::AsyncResource {
   napi_finalize finalize_cb;
   napi_threadsafe_function_call_js call_js_cb;
   bool handles_closing;
+
+  // This variable is set to `true` in the constructor because the constructor
+  // only ever runs on a JS thread, whether that's the Node.js main thread or
+  // one of its worker threads. Since secondary threads never set this variable
+  // but only check its value (in `Push()`) we can use it to determine whether
+  // a given method is running on a JS thread or a secondary thread.
+  static thread_local bool is_js_thread;
 };
 
+thread_local bool ThreadSafeFunction::is_js_thread = false;
```

Comment:
What about this variable that same to above. I’ve a little confused with how Cpp deal with this